Kernel/FileSystem/Ext2Base.h

```diff
@@ -0,0 +1,307 @@
+    u32 create_time; // Inode change time
+    u32 modify_time; // Modification time
+    u32 delete_time; // Deletion Time
+    u16 group_id;    // Low 16 bits of Group Id
+    u16 num_links;   // Links count
+    u32 num_blocks;  // Blocks count
+    u32 file_flags;  // File flags
+
+    // This is Linux dependent
+    u32 linux_version;
+
+    u32 block_ptrs[Ext2::num_blocks]; // Pointers to blocks
+    u32 file_generation;              // File version (for NFS)
+    u32 file_acl;                     // File ACL
+    u32 directory_acl;                // Directory ACL
+    u32 fragment_addr;                // Fragment address
+
+    // These fields below are also Linux dependent.
+    u16 num_blocks_high;
+    u16 file_acl_high;
+    u16 owner_uid_high;
+    u16 group_id_high;
+    u32 reserved2;
+};
+
+// Revision levels
+enum class Revision : u32 {
+    Original, // The good old (original) format
+    Dynamic,  // V2 format w/ dynamic inode sizes
+};
+
+// The structure of the super block
+struct SuperBlock {
+    u32 num_inodes;              // Inodes count
+    u32 num_blocks;              // Blocks count
+    u32 num_reserved_blocks;     // Reserved blocks count
+    u32 num_free_blocks;         // Free blocks count
+    u32 num_free_inodes;         // Free inodes count
+    u32 first_data_block;        // First Data Block
+    u32 block_log_size;          // Block size
+    i32 fragment_log_size;       // Fragment size
+    u32 num_blocks_per_group;    // # Blocks per group
+    u32 num_fragments_per_group; // # Fragments per group
+    u32 num_inodes_per_group;    // # Inodes per group
+    u32 mount_time;              // Mount time
+    u32 write_time;              // Write time
+    u16 num_mounts;              // Mount count
+    i16 num_max_mounts;          // Maximal mount count
+    u16 magic_number;            // Magic signature
+    u16 file_system_state;       // File system state
+    u16 error_behavior;          // Behavior when detecting errors
+    u16 minor_rev_level;         // minor revision level
+    u32 last_check_time;         // time of last check
+    u32 check_interval;          // max. time between checks
+    u32 os_creator;              // OS
+    u32 rev_level;               // Revision level
+    u16 reserved_block_uid;      // Default uid for reserved blocks
+    u16 reserved_block_gid;      // Default gid for reserved blocks
+
+    // These fields are for EXT2_DYNAMIC_REV superblocks only.
+    //
+    // Note: the difference between the compatible feature set and
+    // the incompatible feature set is that if there is a bit set
+    // in the incompatible feature set that the kernel doesn't
+    // know about, it should refuse to mount the filesystem.
+    //
+    // e2fsck's requirements are more strict; if it doesn't know
+    // about a feature in either the compatible or incompatible
+    // feature set, it must abort and not try to meddle with
+    // things it doesn't understand...
```

Comment:
At least this commit appears to be lifted straight from the Linux header, and many other parts seem to be "heavily inspired". IANAL, but I'm questioning whether we can and should relicense the file in that case.

Reply:
imho this can just be dropped, we don't care about the compatibility stuff or e2fsck behavior here